code/datums/datum.dm

```diff
@@ -57,6 +57,15 @@
 			qdel(C, FALSE, TRUE)
 		dc.Cut()
 
+	_clear_signal_refs()
+	//END: ECS SHIT
+
+	return QDEL_HINT_QUEUE
+
+/// Do not override this. This proc exists solely to be overriden by /turf. This
+/// allows it to ignore clearing out signals which refer to it, in order to keep
+/// those signals valid after the turf has been changed.
+/datum/proc/_clear_signal_refs()
```

Comment:
Wondering if there might be some kind of spacemandmm thing we could use to enforce this?